src/back/spv/writer.rs

```diff
@@ -1386,7 +1474,19 @@ impl Writer {
                     let component_id = self.cached[component];
                     constituent_ids.push(component_id);
                 }
-                self.write_composite_construct(result_type_id, &constituent_ids, block)
+                let composite_id =
+                    self.write_composite_construct(result_type_id, &constituent_ids, block);
+
+                // If the composite is an array, emit a store to write
+                // it into an allocated variable
+                if let Some(&var_id) = function.local_const_arrays.get(&expr_handle) {
+                    block
+                        .body
+                        .push(Instruction::store(var_id, composite_id, None));
+                    var_id
```

Comment:
I don't think we can return `var_id` here. The type of the returned SPIR-V is still expected to be a value, not a pointer. It seems to me that we should always return `composite_id` from this block?